.devcontainer/.bashrc

```diff
@@ -0,0 +1,18 @@
+# ~/.bashrc: executed by bash(1) for non-login shells.
```

Comment:
Sorry, some unrelated devcontainer changes leaked into my branch.  Ignore the first few files.

Reply:
...actually...  all the good stuff (@tiny-calc/types changes) are at the bottom.  Maybe start there?

The main idea is to split I*Producer into two parts, one that describes the shape and one that describes the values...  just like the ITree* interfaces.